### PR TITLE
Tests and objects added

### PR DESCRIPTION
**Pull request by EugeneDar**
Closes: #32

---

## PR-Codex overview
This PR adds two new entities to the codebase, Lambda and NestedObject, and includes tests for ObjectsBox class.

### Detailed summary
- Adds Lambda entity class
- Adds NestedObject entity class
- Includes tests for ObjectsBox class, covering Lambda and NestedObject scenarios

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by EugeneDar** [comment]
@Graur please, have a look

**Review by Graur — CHANGES_REQUESTED** [review]

File: src/main/java/org/objectionary/entities/ObjectWithApplication.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ */
+package org.objectionary.entities;
+
+import java.util.Map;
+
+/**
+ * This class represents the object with application entity.
+ * @since 0.1.0
+ */
+public final class ObjectWithApplication extends Entity {
```
[the remark this answers is not in the document]
Comment (EugeneDar) [reply src/main/java/org/objectionary/entities/ObjectWithApplication.java]: @Graur fixed, please have a look again

[the remark this answers is not in the document]
Comment (Graur) [reply src/main/java/org/objectionary/entities/ObjectWithApplication.java]: @EugeneDar Great! You can mark this suggesting as `resolved`

**Review by Graur — APPROVED** [review]
@EugeneDar Good one
@yegor256 Please, have a look

**Comment by EugeneDar** [comment]
@yegor256 please, have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**